src/module.cpp

```diff
@@ -478,28 +454,34 @@ void module::debug_print(instruction_ref ins) const
         return;
     }
     std::stringstream ss;
-    this->print([&](auto x, const auto& names) {
+    this->print([&](auto x, auto ins_names) {
         if(x == ins)
         {
-            print_instruction(std::cout, x, names);
+            instruction::print(std::cout, x, ins_names);
             std::cout << std::endl;
         }
-    });
+    }, names);
+}
+
+void module::debug_print(instruction_ref ins) const
+{
+    std::unordered_map<instruction_ref, std::string> names;
+    this->debug_print(ins, names);
 }
+
 void module::debug_print(const std::vector<instruction_ref>& inss) const
 {
     for(auto ins : inss)
         this->debug_print(ins);
     std::cout << std::endl;
 }
 
-void module::print(const std::function<
+std::unordered_map<instruction_ref, std::string> module::print(const std::function<
                    void(instruction_ref, const std::unordered_map<instruction_ref, std::string>&)>&
-                       print_func) const
+                       print_func,
+                   std::unordered_map<instruction_ref, std::string> names = {}) const
```

Comment:
The default parameter should go in the function definition in the header file, not here.

Reply:
changed